dist/rules/no-internal.js

```diff
@@ -82,21 +89,57 @@ module.exports = {
       return undefined;
     }
 
-    /** @param {string} fileName */
-    function isLocalFile(fileName) {
-      return fileName && typeof fileName === "string" && !fileName.includes("node_modules");
+    function dirContainsPath(dir, targetPath) {
+      const relative = path.relative(dir, targetPath);
+      return (
+        !!relative && !relative.startsWith("..") && !path.isAbsolute(relative)
+      );
+    }
+
+    function pathContainsCheckedPackage(packagePath) {
+      return checkedPackageRegexes.some((r) => r.test(packagePath));
+    }
+
+    function packageNameContainsCheckedPackage(packagePath) {
+      const packageList = workspace.getWorkspaces(packagePath);
+      // Look through all package infos to find the one containing our packagePath
+      for (let pkg in packageList) {
+        const packageObj = packageList[pkg];
+        const packageBaseDir = packageObj.packageJson.packageJsonPath.split("package.json")[0];
+
+        // Note: this returns false if packagePath == packageBaseDir
```

Comment:
can you move this comment to a doc comment on `dirContainsPath`? Indeed the name is confusing if you don't also have `dirContainsOrIsPath` also like some other projects do, but we don't need the latter here